reckless_github/src/repository.rs

```diff
@@ -48,7 +140,11 @@ impl Repository for Github {
         );
         let res = git2::Repository::clone(&self.url, get_dir_path_from_url(&self.url));
         match res {
-            Ok(repo) => clone_recursive_fix(repo, &self.url),
+            Ok(repo) => {
+                let clone = clone_recursive_fix(repo, &self.url);
+                self.index_repository();
```

Comment:
The only error left here is:
```
cannot borrow `*__self` as mutable, as it is behind a `&` reference
`__self` is a `&` reference, so the data it refers to cannot be borrowed as mutable
repository.rs(134, 20): consider changing this binding's type to be: `&mut Github`
```

Reply:
You need to change the self as mut variable is all repository!

This are the diff needed

```diff
diff --git a/reckless_github/src/repository.rs b/reckless_github/src/repository.rs
index d5d750d..9a21468 100644
--- a/reckless_github/src/repository.rs
+++ b/reckless_github/src/repository.rs
@@ -131,7 +131,7 @@ impl Repository for Github {
     ///
     /// Where to store the index is an implementation
     /// details.
-    async fn init(&self) -> Result<(), RecklessError> {
+    async fn init(&mut self) -> Result<(), RecklessError> {
         debug!(
             "INITIALIZING REPOSITORY: {} {} > {}",
             self.name,

```

```diff
diff --git a/reckless_lib/src/repository.rs b/reckless_lib/src/repository.rs
index 8503bac..7959704 100644
--- a/reckless_lib/src/repository.rs
+++ b/reckless_lib/src/repository.rs
@@ -11,7 +11,7 @@ pub trait Repository {
     /// machine.
     ///
     /// This should work like a `git fetch`.
-    async fn init(&self) -> Result<(), RecklessError>;
+    async fn init(&mut self) -> Result<(), RecklessError>;
 
     /// return the list of plugin that are register contained inside the repository.
     async fn list(&self) -> Result<Vec<Plugin>, RecklessError>;

```

run `make` and fix the mutability suggestion